portability-core/src/test/java/org/dataportabilityproject/serviceProviders/flickr/FlickrPhotoServiceTest.java

```diff
@@ -93,6 +106,40 @@ public void toCommonPhoto() {
     assertThat(photoModel.getMediaType()).isEqualTo("image/jpeg");
   }
 
+  @Test
+  public void importStoresAlbumsInJobCache() throws IOException, FlickrException {
+    // Set up input: a single photo album with a single photo
+    PhotosModelWrapper wrapper = new PhotosModelWrapper(Arrays.asList(PHOTO_ALBUM),
+        Arrays.asList(PHOTO_MODEL), new ContinuationInformation(null, null));
+
+    // Set up mocks
+    when(imageStreamProvider.get(FETCHABLE_URL)).thenReturn(inputStream);
+
+    when(uploader
+        .upload(Matchers.any(BufferedInputStream.class), Matchers.any(UploadMetaData.class)))
+        .thenReturn(FLICKR_PHOTO_ID);
+
+    String flickrAlbumTitle = FlickrPhotoService.FLICKR_ALBUM_PREFIX + ALBUM_NAME;
+    Photoset photoSet = initializePhotoset(FLICKR_ALBUM_ID, flickrAlbumTitle, ALBUM_DESCRIPTION);
+    when(photosetsInterface.create(flickrAlbumTitle, ALBUM_DESCRIPTION, FLICKR_PHOTO_ID))
+        .thenReturn(photoSet);
+
+    // Run test
+    photoService.importItem(wrapper);
+
+    // Verify the mocks were called correctly
+    verify(imageStreamProvider).get(FETCHABLE_URL);
+    verify(uploader)
+        .upload(Matchers.any(BufferedInputStream.class), Matchers.any(UploadMetaData.class));
```

Comment:
Not sure if you're aware, another cool thing you can do with mockito is ArgumentCaptors. In this case I think it'd be useful to capture the uploadMetadata that is passed to uploader, and verify the title is what you expect.

Reply:
Sweet!

Reply:
Nice! So for the other fields, this may be debatable but I don't think checking the values of those flags (isPublic etc) adds much value because they are hardcoded to false, whereas with the title we are checking some (albeit simple) logic from the class, i.e. the string concatenation and that the correct photo is passed.

Checking the other flag values might be like a change-detector test a la TOTT